src/main.rs

```diff
@@ -1097,6 +1097,72 @@ mod tests {
         });
     }
 
+    #[test]
+    fn test_vhost_user_net() {
+        test_block!(tb, "", {
+            let mut clear = ClearDiskConfig::new();
+            let guest = Guest::new(&mut clear);
+
+            let mut workload_path = dirs::home_dir().unwrap();
+            workload_path.push("workloads");
+
+            let mut vubridge_path = workload_path.clone();
+            vubridge_path.push("vubridge");
+            let vubridge_path = String::from(vubridge_path.to_str().unwrap());
+
+            // Start the daemon
+            let mut daemon_child = Command::new(vubridge_path.as_str()).spawn().unwrap();
+
+            let mut cloud_child = Command::new("target/debug/cloud-hypervisor")
+                .args(&["--cpus", "4"])
+                .args(&["--memory", "size=512M,file=/dev/shm"])
+                .args(&["--kernel", guest.fw_path.as_str()])
+                .args(&[
+                    "--disk",
+                    guest
+                        .disk_config
+                        .disk(DiskType::OperatingSystem)
+                        .unwrap()
+                        .as_str(),
+                    guest
+                        .disk_config
+                        .disk(DiskType::CloudInit)
+                        .unwrap()
+                        .as_str(),
+                ])
+                .args(&[
+                    "--vhost-user-net",
+                    "mac=52:54:00:02:d9:01,sock=/tmp/vubr.sock",
+                ])
+                .args(&["--net", guest.default_net_string().as_str()])
+                .spawn()
+                .unwrap();
+
+            thread::sleep(std::time::Duration::new(5, 0));
+            let _ = daemon_child.kill();
+            let _ = daemon_child.wait();
```

Comment:
While guest connects to the socket provided by the daemon process, it will keep printing the transfer network package content, I really don't want to make the integration console log full of it, once ensure it works normally, just kill it.

Reply:
You mean to add additional tests for connection between daemon process (vhost-user-bridge) and guest? Hmm, so far, from the implementation of vhost-user-bridge, my understanding is it's just created to test vhost-user connection. What type of functional tests you'd like to see?

Reply:
@bjzhjing @sameo @rbradford 
Ok so I spent today going through testing of vhost-user-net. I found out several ways we can test end to end network using your implementation + `vubridge` daemon. Please take a look at the [document](https://gist.github.com/sboeuf/b44da3c1fd3335ddfd45e33be67085a5) I wrote about this. The idea is to really test we have network packets flowing properly through our vhost-user-net implementation.
I wanted to go and implement one of the two methods I explain in the document, but I realized we needed `netcat` (`nc` binary) to perform all this testing. Unfortunately, `nc` is not part of the CL cloud image. We need to properly install `nc` through `swupd` and then push this updated image to our cloud-hypervisor storage account on Azure.

I would suggest that we can merge the PR for now with this basic testing (making sure the amount of network interfaces being created is as expected).
In the meantime, @rbradford could you take care of the image update? If you don't have time for this, please let me know.
Once the image will be updated, I'll work on extending the test.

@bjzhjing one last thing, I would really prefer if you could kill the `daemon_child` after you verified the amount of network interfaces.